presto-parquet/src/main/java/com/facebook/presto/parquet/readerv2/Int32SimpleColumnReaderV2.java

```diff
@@ -0,0 +1,154 @@
+        extends AbstractSimpleColumnReaderV2
+{
+    public Int32SimpleColumnReaderV2(RichColumnDescriptor columnDescriptor)
+    {
+        super(columnDescriptor);
+    }
+
+    @Override
+    protected ColumnChunk readWithNull()
+            throws IOException
+    {
+        int[] values = new int[nextBatchSize];
+        boolean[] isNull = new boolean[nextBatchSize];
+
+        int remainingInBatch = nextBatchSize;
+        int startOffset = 0;
+        while (remainingInBatch > 0) {
+            if (remainingCountInPage == 0) {
+                if (!readNextPage()) {
+                    break;
+                }
+            }
+
+            int readChunkSize = Math.min(remainingCountInPage, remainingInBatch);
+            int nonNullCount = dlDecoder.readNext(isNull, startOffset, readChunkSize);
+
+            ((Int32ValuesDecoder) valuesDecoder).readNext(values, startOffset, nonNullCount);
+
+            int valueDestIdx = startOffset + readChunkSize - 1;
+            int valueSrcIdx = startOffset + nonNullCount - 1;
+
+            while (valueDestIdx >= startOffset) {
+                if (!isNull[valueDestIdx]) {
+                    values[valueDestIdx] = values[valueSrcIdx];
+                    valueSrcIdx--;
+                }
+                valueDestIdx--;
+            }
+
+            startOffset += readChunkSize;
+            remainingInBatch -= readChunkSize;
+            remainingCountInPage -= readChunkSize;
+        }
+
+        if (remainingInBatch != 0) {
+            throw new ParquetDecodingException("Still remaining to be read in current batch.");
+        }
+
+        Block block = new IntArrayBlock(nextBatchSize, Optional.of(isNull), values);
+        return new ColumnChunk(block, new int[0], new int[0]);
+    }
+
+    @Override
+    protected ColumnChunk readNoNull()
+            throws IOException
+    {
+        int[] values = new int[nextBatchSize];
+
+        int remainingInBatch = nextBatchSize;
+        int startOffset = 0;
+        while (remainingInBatch > 0) {
+            if (remainingCountInPage == 0) {
+                if (!readNextPage()) {
+                    break;
+                }
+            }
+
+            int readChunkSize = Math.min(remainingCountInPage, remainingInBatch);
+
+            Int32ValuesDecoder valuesDecoder = (Int32ValuesDecoder) this.valuesDecoder;
+            valuesDecoder.readNext(values, startOffset, readChunkSize);
+
+            startOffset += readChunkSize;
+            remainingInBatch -= readChunkSize;
+            remainingCountInPage -= readChunkSize;
+        }
+
+        if (remainingInBatch != 0) {
+            throw new ParquetDecodingException("Still remaining to be read in current batch.");
+        }
+
+        Block block = new IntArrayBlock(nextBatchSize, Optional.empty(), values);
+        return new ColumnChunk(block, new int[0], new int[0]);
+    }
+
+    @Override
+    protected void skip(int skipSize)
+            throws IOException
+    {
```

Comment:
int remainingInBatch = skipSize;
and remove the inner definitions in both if cases